dali/c_api/c_api.h

```diff
@@ -31,6 +33,22 @@ extern "C" {
     GPU = 1
   };
 
+  typedef enum {
+    DALI_NO_TYPE         = -1,
+    DALI_UINT8           =  0,
+    DALI_UINT16          =  1,
+    DALI_UINT32          =  2,
+    DALI_UINT64          =  3,
+    DALI_INT8            =  4,
+    DALI_INT16           =  5,
+    DALI_INT32           =  6,
+    DALI_INT64           =  7,
+    DALI_FLOAT16         =  8,
+    DALI_FLOAT           =  9,
+    DALI_FLOAT64         = 10,
+    DALI_BOOL            = 11
+  } dali_data_type_t;
```

Comment:
Cab we make sure it is strongly bound to DALIDataType values?

Reply:
I don't think so unless we refactor the DALIDataType to some C-friendly header.